src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -518,7 +518,7 @@ internal void Initialize(
 
             var analyzerExecutor = AnalyzerExecutor.Create(
                 compilation, options, addNotCategorizedDiagnostic, newOnAnalyzerException, analysisOptions.AnalyzerExceptionFilter,
-                IsCompilerAnalyzer, analysisScope.Analyzers, analysisOptions.GetAnalyzerConfigOptionsProvider,
+                IsCompilerAnalyzer, this.Analyzers, analysisOptions.GetAnalyzerConfigOptionsProvider,
```

Comment:
the core issue here is that the compWithAnalyzer is created with a larger set of analyzers than the scope we're asking for results for.  However, both the analyzers we're asking for results for *and* the diagnostic supressors given to the CompWithAnalyzers will lookup in the diagnosticOptions map to see how they are configured.  Because we weren't passing in all analyzers here, we weren't setting up the map properly, and would crash trying to index into it with the DIagnosticSuppressor.

Reply:
Another option would be to use analysisScope.Analyzers and concat any diagnostic supressors in this.Analyzers.  But this is really fine to just ensure we set up the map for any purpose.